src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule73wherejavadocrequired/InputMissingJavadocTypeCheck.java

```diff
@@ -0,0 +1,45 @@
+package com.google.checkstyle.test.chapter7javadoc.rule73wherejavadocrequired;
+
+public class InputMissingJavadocTypeCheck { //warn
+
+    public class Inner { // warn
+
+    }
+
+    public enum MyEnum { // warn
+
+    }
+
+    public interface MyInterface {  // warn
+        class MyInterfaceClass {}  // warn
+    }
+
+    public @interface MyAnnotation { // warn
+
+    }
+
+    protected class InnerProtected { // warn
+
+    }
+
+    protected enum MyEnumProtected { // warn
+
+    }
+
+    protected interface MyInterfaceProtected {  // warn
+
+    }
+
+    protected @interface MyAnnotationProtected { //warn
```

Comment:
Please add private and package-private fields

Package-private are very similar to protected to some extent.

Reply:
private field is not allowed here:

```
➜  rule73wherejavadocrequired git:(googlestyle-rule73) ✗ cat InputMissingJavadocTypeCheck.java 
package com.google.checkstyle.test.chapter7javadoc.rule73wherejavadocrequired;

public class InputMissingJavadocTypeCheck { //warn
    private int field;

    public class Inner { // warn

    }

    public enum MyEnum { // warn

    }

    public interface MyInterface {  // warn
        class MyInterfaceClass {}  // warn
    }

    public @interface MyAnnotation { // warn

    }

    protected class InnerProtected { // warn

    }

    protected enum MyEnumProtected { // warn

    }

    protected interface MyInterfaceProtected {  // warn

    }

    protected @interface MyAnnotationProtected { //warn
        int field = 0;
        private int x = 0;

    }

    public void myMethod() {
        class MyMethodClass {} // ok
    }

}

class AdditionalClass { // OK, not public

}
➜  rule73wherejavadocrequired git:(googlestyle-rule73) ✗ javac InputMissingJavadocTypeCheck.java
InputMissingJavadocTypeCheck.java:36: error: modifier private not allowed here
        private int x = 0;
                    ^
1 error
➜  rule73wherejavadocrequired git:(googlestyle-rule73) ✗ 

```

Reply:
my commend is misleading, I meant in top class.